client/blocks/author-compact-profile/index.jsx

```diff
@@ -29,14 +30,20 @@ const AuthorCompactProfile = React.createClass( {
 			return null;
 		}
 
+		const hasMatchingAuthorAndSiteNames = siteName.toLowerCase() === author.name.toLowerCase();
```

Comment:
I'm not sure we can safely assume `author.name` is not empty; I just encountered an error `Cannot read property 'toLowerCase' of undefined` at this line when trying to load a detail view for a post in my Reader.